src/sGLOH2.cpp

```diff
@@ -214,42 +219,100 @@ cv::Mat sGLOH2::computeCustomHistogram(const cv::Mat& data, const std::vector<fl
     return histogramMat;
 }
 
-cv::Mat sGLOH2::computeHistogram(const cv::Mat& region, int m) {
-    // Compute the gradient of the region. The Sobel function is used to find the intensity
-    // gradient of the image. For each pixel, a 2D spatial gradient measurement is computed
-    // for an image, which gives the orientation and magnitude of the gradient at each point.
+
+cv::Mat sGLOH2::computeHistogram(const cv::Mat& patch, const cv::Mat& mask, int m) {
+    // Compute the gradient of the region.
     cv::Mat grad_x, grad_y;
-    cv::Sobel(region, grad_x, CV_32F, 1, 0, 3);
-    cv::Sobel(region, grad_y, CV_32F, 0, 1, 3);
+    cv::Sobel(patch, grad_x, CV_32F, 1, 0, 3);
+    cv::Sobel(patch, grad_y, CV_32F, 0, 1, 3);
 
-    // Compute the magnitude and orientation of the gradient. The cartToPolar function
-    // calculates the magnitude and angle of 2D vectors. The angle is computed in radians
-    // and the magnitude is computed as sqrt(x^2 + y^2).
+    // Compute the magnitude and orientation of the gradient.
     cv::Mat magnitude, orientation;
     cv::cartToPolar(grad_x, grad_y, magnitude, orientation, true);
 
-    // Convert orientation from radians to degrees. This is done because the histogram
-    // function expects input in degrees, not radians.
+    // Convert orientation from radians to degrees.
     orientation = orientation * 180.0 / CV_PI;
 
-    // Compute the histogram of orientations. The calcHist function calculates the
-    // histogram of a set of arrays. It can operate on arrays of arbitrary dimensions
-    // and depth. In this case, it's used to compute the histogram of gradient orientations.
+    // Compute the histogram of orientations.
     cv::Mat histogram;
-    int histSize = m;  // Number of bins
-    float range[] = { 0, 360 } ;  // Orientation ranges from 0 to 360 degrees
+    int histSize = m;  // Number of bins.
```

Comment:
Is the number of bins necessarily tied to the number of segments in a ring (m) ? It doesn't necessarily need to be the same right?

Reply:
I guess in their example in the paper it is....but I'm still not sure if it actually needs to be? 🤔

Reply:
Tried a few values 4 worked best